powertools-e2e-tests/src/test/java/software/amazon/lambda/powertools/ValidationE2ET.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2023 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package software.amazon.lambda.powertools;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static software.amazon.lambda.powertools.testutils.Infrastructure.FUNCTION_NAME_OUTPUT;
+import static software.amazon.lambda.powertools.testutils.lambda.LambdaInvoker.invokeFunction;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Timeout;
+import org.junit.jupiter.params.ParameterizedTest;
+
+import com.amazonaws.services.lambda.runtime.events.APIGatewayProxyRequestEvent;
+import com.amazonaws.services.lambda.runtime.tests.annotations.Event;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import software.amazon.lambda.powertools.testutils.Infrastructure;
+import software.amazon.lambda.powertools.testutils.lambda.InvocationResult;
+
+class ValidationE2ET {
+
+	private static final ObjectMapper objectMapper = new ObjectMapper();
+
+	private static Infrastructure infrastructure;
+	private static String functionName;
+
+	@BeforeAll
+	@Timeout(value = 5, unit = TimeUnit.MINUTES)
+	public static void setup() {
+		infrastructure = Infrastructure.builder().testName(ValidationE2ET.class.getSimpleName())
+				.pathToFunction("validation").build();
+		Map<String, String> outputs = infrastructure.deploy();
+		functionName = outputs.get(FUNCTION_NAME_OUTPUT);
+	}
+
+	@AfterAll
+	public static void tearDown() {
+		if (infrastructure != null) {
+			infrastructure.destroy();
+		}
+	}
+
+	@ParameterizedTest
+	@Event(value = "/validation/valid_api_gw_in_out_event.json", type = APIGatewayProxyRequestEvent.class)
+	void test_validInboundApiGWEvent(APIGatewayProxyRequestEvent validEvent) throws IOException {
+		// WHEN
+		InvocationResult invocationResult = invokeFunction(functionName, objectMapper.writeValueAsString(validEvent));
```

Comment:
you actually deserialize with the `@Event` annotation and reserialize here, you can probably just load the file as string to avoid this double step...

Reply:
I can replace loading the event with an InputStream. The following works as well.

`@Test
	void test_validInboundApiGWEvent() throws IOException {
		InputStream inputStream = this.getClass().getResourceAsStream("/validation/valid_api_gw_in_out_event.json");
		String validEvent = IOUtils.toString(inputStream, StandardCharsets.UTF_8);

		// WHEN
		InvocationResult invocationResult = invokeFunction(functionName, validEvent);
`